kubernetes/crd/dataset.json

```diff
@@ -0,0 +1,35 @@
+{
+	"apiVersion": "apiextensions.k8s.io/v1",
+	"kind": "CustomResourceDefinition",
+	"metadata": { "name": "datasets.new.artificialwisdom.cloud" },
+	"spec": {
+		"group": "new.artificialwisdom.cloud",
+		"versions": [{
+			"name": "v1",
+			"served": true,
+			"storage": true,
+			"schema": {
+				"openAPIV3Schema": {
+					"type": "object",
+					"properties": {
+						"spec": {
+							"type": "object",
+							"properties": {
+								"source": { "type": "string" },
+								"filename": { "type": "string" },
+								"bucket": { "type": "string" }
+							}
+						}
+					}
+				}
+			}
+		}],
+		"scope": "Namespaced",
```

Comment:
I feel conflicted about this. I could see a model where Datasets are not namespaced until referenced by a user, and then become namespaced. Not sure if that works, I am beat, so I am going to bed.

Reply:
The general guideline is to think of namespaces as virtual clusters. If we had multiple teams working on multiple products each with their own virtual-cluster isolation, then do we want to let them share resources?

In this case, I think that the answer is "no", and my justification is that we would want disjoint buckets for our different teams and products, so there wouldn't be any overlap anyway. We can talk about it later.